src/components/data/data-utils.js

```diff
@@ -937,6 +937,90 @@ export const MultipleEntityEditor = ({ entityType, entityNames, attributeNames,
   ])
 }
 
+export const CreateEntitySetModal = ({ entityType, entityNames, workspaceId: { namespace, name: workspaceName }, onDismiss, onSuccess }) => {
+  const [name, setName] = useState('')
+  const [nameInputTouched, setNameInputTouched] = useState(false)
+  const nameError = nameInputTouched && Utils.cond(
+    [!name, () => 'A name for the set is required.'],
+    [!/^[A-Za-z0-9_-]+$/.test(name), () => 'Set name may only contain alphanumeric characters, underscores, dashes, and periods.']
+  )
+
+  const [isBusy, setIsBusy] = useState()
+
+  const createSet = async () => {
+    try {
+      setIsBusy(true)
+
```

Comment:
Nit: Would it be more clear what we think may error with the following?

```suggestion
    setIsBusy(true)
    try {
```

Reply:
That makes sense.